### DISCUSSION TITLE
timelapse and hindsight

### DISCUSSION THREAD

**Discussion by Fougl** [gopro/labs#488] [Feature Requests]
Hi, I am wondering if this will ever be possible, to have timelapse mode and hindsight at the same time? Thank you

**Comment by dnewman-gpsw** [comment]
I'm not sure what benefit is gained to have 15-30s (real-time) of pre-roll for a timelapse recording, and 15-30s of timelapse recording is was too much for a preroll.  Also to press the shutter part way into a timelapse is risky, as you can move the camera.  However you can script this yourself with the new blockly Labs scripting tool (I need a name for this)  https://gopro.github.io/labs/build/

**Comment by Fougl** [comment]
I would actually use it with motion detection. I would like to get photos from it, because making videos and then extracting frames would be too expensive for data because I am downloading and then uploading these videos with raspberry. router is with lte sim card and signal is not too good. RPI extract frames with decent speed but still I prefer using RPI zero w which does not have enough computing power

In certain environment settings, I need to use it with hindsight (preferrably 5s, which is available with the new firmwares), but not always. So basically what I am asking is, if hindsight is also available with the same frame frequency as time lapse. So, not real time. Becuase I only need to get a few frames from the 5s hindsight, not the whole video.

**Reply by dnewman-gpsw** [reply]
Motion detection does work with timelapse, but there is no way to combine all three: hindsight, motion detection and timelapse.  As hindsight is not the solution, it seems you only want to transfer particular frames, would it matter if the camera recorded extra frames if you didn't transfer them?  I don't yet have a solution, but maybe something can be worked out. Are you using Open GoPro via RPI?

**Comment by Fougl** [comment]
Yes, I need those hindsight seconds, because the motion detection uses a mask, to avoid unnecessary movements (there are many - it is such environment). So only when a person comes in the part of the frame with the mask, the camera starts recording. But I need to record this person also, before the part with the mask. So hindsight does solve this. But as I said, i dont need the video, i just need like 1 frame per second from the whole recording. So time lapse would really help, not to upload the files and use valuebale time and data

Yes, I am using Open Gopro to first download the video and then upload to cloud.

**Reply by Fougl** [reply]
Hi, Any more thoughts on this one? Do you think it would be possible to have hindsight in time lapse mode? Or gopro does not allow this? This would save my life

**Reply by dnewman-gpsw** [reply]
No don't think it will be possible.  It is not part of hindsight. so Labs can't add it.
